OMG.Classroom/DataAccessLayer/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using DataAccessLayer.Entities;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Repositories
+{
+    public abstract class BaseRepository<E> where E : BaseEntity
+    {
+        public readonly ClassroomDbContext _context;
+        public readonly DbSet<E> _dbSet;
+
+        protected BaseRepository(ClassroomDbContext context, DbSet<E> dbSet)
```

Comment:
You can set the DbSet directly in the constrcutor, no need to pass it as an argument